src/main/java/uk/gov/hmcts/reform/pdf/service/aspect/AuthAspect.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.pdf.service.aspect;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.pdf.service.service.AuthService;
+
+@Aspect
+@Component
+public class AuthAspect {
+
+    @Autowired
+    private AuthService authService;
+
+    @Before("execution(* uk.gov.hmcts.reform.pdf.service.endpoint.v2.*.*(String, ..))")
+    public void authenticate(JoinPoint jp) {
```

Comment:
not sure if its better but I would normally do request intercepting not aspectj for this sort of thing.
https://jmnarloch.wordpress.com/2015/10/14/spring-cloud-feign-oauth2-authentication/

Anyone have any thoughts?

Reply:
Not strong on this as I cannot find an argument why aspects are wrong, but I agree with @timja-kainos